libraries/wallet/wallet.cpp

```diff
@@ -3012,6 +3133,67 @@ uint64_t wallet_api::get_asset_count()const
    return my->_remote_db->get_asset_count();
 }
 
+std::vector<unsigned char> string_to_vec(std::string incoming)
+{
+   char s[3];
+   s[2] = 0;
+   std::vector<unsigned char> vec;
+   for(int i = 0; i < incoming.length(); i+= 2)
+   {
+      s[0] = incoming[i];
+      s[1] = incoming[i+1];
+      vec.push_back( (int)strtol(&s[0], nullptr, 16));
+   }
+   return vec;
+}
+
+signed_transaction wallet_api::htlc_prepare( string source, string destination, string asset_symbol, string amount,
+         string hash_algorithm, const std::string& preimage_hash, size_t preimage_size, 
+         const uint32_t seconds_in_force, bool broadcast)
+{
+   return my->htlc_prepare(source, destination, asset_symbol, amount, hash_algorithm, string_to_vec(preimage_hash), preimage_size,
+         seconds_in_force, broadcast);
+}
+
+std::string object_id_to_string(const graphene::db::object_id_type& obj_id) 
```

Comment:
Can use one of these:
* https://github.com/bitshares/bitshares-core/blob/2.0.181127/libraries/db/include/graphene/db/object_id.hpp#L83
* https://github.com/bitshares/bitshares-core/blob/2.0.181127/libraries/db/include/graphene/db/object_id.hpp#L192